account_asset_management/migrations/12.0.1.0.0/post-migration.py

```diff
@@ -18,20 +18,44 @@ def adjust_asset_values(env):
         FROm account_asset_profile aap
         WHERE aa.profile_id = aap.id""",
     )
-    # Adjust method_time, method_number and method_period
+    # Adjust method_time, method_end, method_number and method_period
     number = sql.Identifier(openupgrade.get_legacy_name('method_number'))
     period = sql.Identifier(openupgrade.get_legacy_name('method_period'))
+    time = sql.Identifier(openupgrade.get_legacy_name('method_time'))
     for table in ['account_asset_profile', 'account_asset']:
         table = sql.Identifier(table)
         openupgrade.logged_query(
             env.cr, sql.SQL("""
             UPDATE {table}
             SET method_time = 'year',
+                method_end = NULL,
                 method_number = ({number} * {period}) / 12
-            WHERE MOD({number} * {period}, 12) = 0
+            WHERE MOD({number} * {period}, 12) = 0 AND {time} != 'end'
             """).format(
                 number=number,
                 period=period,
+                time=time,
```

Comment:
You should call `method_time` for clarity

Reply:
The same happens with number and period...

Reply:
Yes, of course. Put good names to all.